### PR TITLE
Transactional Failure: Display failed transactions in the activity list

### LINKED ISSUE #587: [Bug]: Txn details page - missing gas
### Discord Discussion Link

_No response_

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

View txn details

### What did not work?

Gas is populating as 'unknown' (this is an hour after adding account)

Screen Shot 2021-12-11 at 5 49 16 PM

### Version

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #678: Properly display failed transactions
As discovered testing a [recent PR](https://github.com/tallycash/extension/pull/673), we aren't showing failed transactions in activities! Here's what a failed tx currently looks like

Screenshot from 2021-12-15 22-26-50

Instead, it should look... failed, and show the gas that was expended rather than the transaction value which didn't do anything.

### PR DESCRIPTION
**Pull request by mhluongo**
Screenshot from 2021-12-16 16-19-00

Does what it says on the tin! Displays when transactions fail in the activity list.

Closes #678, closes #587.

### CODE REVIEW

File: background/services/chain/index.ts [hunk 1]
```diff
@@ -592,6 +592,8 @@ export default class ChainService extends BaseService<Events> {
         } else if (transaction.blockHash) {
           // Get relevant block data.
           await this.getBlockData(transaction.network, transaction.blockHash)
+          // Retrieve gas used, status, etc
+          this.retrieveTransactionReceipt(transaction.network, transaction)
```
Comment (Shadowfiend) [inline background/services/chain/index.ts:596]: I hate how none of our things that save say that they do in their names, fwiw. Would prefer a `fetchAndSave` or something. It's systemic though, so not really for this PR.

File: background/services/chain/index.ts [hunk 2]
```diff
@@ -741,7 +743,7 @@ export default class ChainService extends BaseService<Events> {
    * the database and informing subscribers via the emitter.
    *
    * @param network the EVM network we're interested in
-   * @param txHash the hash of the unconfirmed transaction we're interested in
+   * @param transaction the unconfirmed transaction we're interested in
```
Comment (Shadowfiend) [inline background/services/chain/index.ts:746]: :grimacing:

File: ui/components/Wallet/WalletActivityListItem.tsx [hunk 3]
```diff
@@ -105,6 +105,11 @@ export default function WalletActivityListItem(props: Props): ReactElement {
               className={classNames("activity_icon", renderDetails.iconClass)}
             />
             {renderDetails.label}
+            {"status" in activity && activity.status !== 1 ? (
+              <div className="failed">Failed</div>
```
Comment (Shadowfiend) [inline ui/components/Wallet/WalletActivityListItem.tsx:109]: We should really add a `status` class for `failed` and `pending` and do `.status:before` in CSS, letting the `failed`/`pending` classes just apply the `color`.

We should probably also at some point give `ActivityItem` a more delineated `status` field that carries `pending`/`failed`/`succeeded`/whatever else in one field, populated in the selectors.

**Review by Shadowfiend — COMMENTED** [review]
Code looks good, test driving now.

**Review by Shadowfiend — APPROVED** [review]
Does what it says on the tin indeed!

**Comment by Shadowfiend** [comment]
Actually does more than it says on the tin, added another ticket close to the description 😏

File: ui/components/Wallet/WalletActivityListItem.tsx (hunk 3 above)
Comment (mhluongo) [reply ui/components/Wallet/WalletActivityListItem.tsx:109]: From a sync discussion: we have a similar issue with `timestamp` and other stuff that's sort of transaction data, sort of "joined" network data (in these cases with the block and tx receipt). The choice here was to stay closer to the EVM with that `number`... but we'll want to reconsider `EVMTransaction` soon.

Comment (mhluongo) [reply ui/components/Wallet/WalletActivityListItem.tsx:109]: ... and yes, `ActivityItem` needs to lift heavier.

**Pull request merged**